flake.nix

```diff
@@ -0,0 +1,51 @@
+{
+  inputs = {
+    nixpkgs.url = "github:serokell/nixpkgs";
+    haskell-nix.url = "github:input-output-hk/haskell.nix";
+    hackage = {
+      url = "github:input-output-hk/hackage.nix";
+      flake = false;
+    };
+    stackage = {
+      url = "github:input-output-hk/stackage.nix";
+      flake = false;
+    };
+    common-infra.url = "github:serokell/common-infra";
+  };
+
+  outputs = { self, nixpkgs, haskell-nix, hackage, stackage, common-infra }:
+    let
+      packagesFor = system: rec {
+        pkgs = nixpkgs.legacyPackages.${system}.extend
```

Comment:
What is the semantic of this `pkgs`?

Reply:
Flakes define derivations for all known systems. In the context of a flake, `nixpkgs` isn't a set of packages. It's a set of sets of packages. So you need to use the set for the current (for some definition of "current") system.